### ISSUE TITLE
Warning when compiling a page

### ISSUE THREAD

**Issue by tuhlmann** [kentcdodds/old-kentcdodds.com#92]
Hey Kent,

first of all a huge Thank You for opening up this blog and giving us opportunity to learn and reuse!

I see a warning in your code that I can't quite put my fingers on. When running `yarn dev` and edit a page so it gets updated in the browser, I see this error:

```
info changed file at /d/entw/www/kentcdodds.com/content/blog/react-strict-mode/index.md
ℹ ｢wdm｣: Compiling...
warning Action createPage was called outside of its expected asynchronous lifecycle createPages in default-site-plugin.
Ensure that you return a Promise from createPages and are awaiting any asynchronous method invocations (like graphql or http requests).
For more info and debugging tips: see https://gatsby.app/sync-actions

File /d/entw/www/kentcdodds.com/gatsby-node.js:25:5
  23 |     }
  24 |
> 25 |     createPage({
     |     ^
  26 |       path: pagePath,
  27 |       component: path.resolve(`./src/templates/post.js`),
  28 |       context: {
```

As far as I can tell the function is called in an async context, I'm not sure what's missing. The update works fine, it's just that message that appears.
Is this just a weird combination of HMR and gatsby compiler interfering?

Thanks again,
Torsten.

**Comment by kentcdodds** [comment]
Hey @tuhlmann!

That's actually a false warning. It's a bug in gatsby that will hopefully eventually be fixed. cc @DSchau

I'm also waiting to upgrade my version of gatsby until a more serious bug is fixed.

**Comment by DSchau** [comment]
I think we actually fixed in gatsbyjs/gatsby#11929, available in gatsby@2.1.14.

Since we have it pinned here, unable to get the upstream fix. Let me check and see if I can reproduce the error re: hot reloaded content not appearing and if so, will add that to an existing issue.

Thanks for reporting this @tuhlmann and for the cc @kentcdodds! We'll take a look.

**Comment by tuhlmann** [comment]
Awesome, thanks! One more little thing that makes more sense now :)
